beps/0001-notifications-system/README.md

```diff
@@ -85,12 +85,43 @@ The role of the notifications plugin is to manage the lifecycle of notifications
 The notification backend stores notification using the [database service](https://backstage.io/docs/backend-system/core-services/index#database). In particular it needs to store the following information for each notification:
 
 - ID
-- Payload
+- Title
+- Priority
+- Origin
+- Topic (optional)
+- Creation timestamp (generated)
 - Receivers
 - Read status by user
+- Payload
+  - icon (optional)
+  - message (optional)
+  - links (list, optional)
 
 The receivers is **not** a list of users, but rather a filter that describes who should receive the notification. It must be possible to evaluate this filter in a database query, so that we can efficiently fetch all notifications for a given user. The same filter will also be used by the signal backend to determine which users should receive a signal.
 
+The title is mandatory human readable text shortly describing the notifications.
+
+The priority is one of the predefined values to help with presentation of the notifications or their filtering. Values: Critical, High, Normal, Low.
+
+The origin is a string identifying the creator of the notification, i.e. the BE plugin or external system.
+
+The topic is an optional string helping to group notifications of particular context. Its use cases include:
+
+- several notifications emitted by an asynchronous external task can be grouped by a single topic
+- a BE plugin can group several related messages to a particular processing, i.e. asynchronous progress monitoring
+
+The timestamp of message creation is auto-generated by the notifications backend at the time of receiving a request to create the notification.
+
+The icon is optional and is meant to improve UX. A string referencing an icon name from MUI icon library of a defined version. If missing, it will be determined from the priority
+
+The message is an optional human readable text providing more details to the user.
+
+The links are an array of title-URL pairs. As an example, they can be used
```

Comment:
Initial thought here is that a single "main" link might be better, and any additional links end up being rendered by custom renderers in the frontend.

Reply:
Can it be done when the message is originated in an external system?
Example:
- An async process did not go well. The user is asked for further info (one link) and is provided with logs or monitoring (second link).
- An async process come to a yes-no user question (two links)

I believe we can create more examples where multiple links would be handy.